test/Microsoft.Sbom.Api.Tests/Executors/FileInfoWriterTests.cs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.Sbom.Api.Entities;
+using Microsoft.Sbom.Api.Executors;
+using Microsoft.Sbom.Api.Manifest;
+using Microsoft.Sbom.Contracts;
+using Microsoft.Sbom.Contracts.Enums;
+using Microsoft.Sbom.Extensions;
+using Microsoft.Sbom.Extensions.Entities;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Serilog;
+using Constants = Microsoft.Sbom.Api.Utils.Constants;
+
+namespace Microsoft.Sbom.Api.Tests.Executors;
+
+[TestClass]
+public class FileInfoWriterTests
+{
+    private Mock<ILogger> loggerMock;
+    private Mock<ISbomConfig> sbomConfigMock;
+    private Mock<ISbomPackageDetailsRecorder> sbomPackageDetailsRecorderMock;
+    private Mock<IManifestToolJsonSerializer> manifestToolJsonSerializerMock;
+    private FileInfoWriter testSubject;
+    private ManifestGeneratorProvider manifestGeneratorProvider;
+
+    [TestInitialize]
+    public void BeforeEach()
+    {
+        loggerMock = new Mock<ILogger>();
+        sbomPackageDetailsRecorderMock = new Mock<ISbomPackageDetailsRecorder>();
+        manifestToolJsonSerializerMock = new Mock<IManifestToolJsonSerializer>();
+
+        manifestGeneratorProvider = new ManifestGeneratorProvider(new IManifestGenerator[] { new TestManifestGenerator() });
+        manifestGeneratorProvider.Init();
+
+        testSubject = new FileInfoWriter(manifestGeneratorProvider, loggerMock.Object);
+
+        sbomConfigMock = new Mock<ISbomConfig>();
+        sbomConfigMock.SetupGet(x => x.ManifestInfo).Returns(Constants.TestManifestInfo);
+        sbomConfigMock.SetupGet(x => x.Recorder).Returns(sbomPackageDetailsRecorderMock.Object);
+        sbomConfigMock.SetupGet(x => x.JsonSerializer).Returns(manifestToolJsonSerializerMock.Object);
+    }
+
+    [TestCleanup]
+    public void AfterEach()
+    {
+        // Just verify nothing throws
+    }
+
+    [TestMethod]
+    public async Task Write_FileWithinDropPath_WritesToFilesSection()
+    {
+        // Arrange
```

Comment:
With the exception of the `Path` and the `CheckSum`, lines 59-70 seem to mirror lines 101-112. Would this be a good candidate for a helper method to set things up? You might need to make `FileChannelInfo` and `sbomConfigs` class members, or else pass them back from the helper method.